lib/exercism/user_contribution.rb

```diff
@@ -0,0 +1,64 @@
+require_relative '../../x'
+class UserContribution
+  attr_reader :user
+  def initialize(user)
+    @user = user
+  end
+
+  # Return Hahs with, name of the type of contribution, also the erb partial and
+  # the info related with each contributions
+  def list
+    @list ||= { completed_tracks: completed_tracks }
+  end
+
+  def any?
+    list.any? { |_, v| !v.empty? }
+  end
+
+  private
+
+  def completed_tracks
+    completed_exercises_by_language.select do |hash|
+      hash['completed_exercise'].to_i == hash['total_in_track']
+    end
+  end
+
+  # Hit the database, and return a hash containing, valuable information, about
+  # the user tracks.
+  # [
+  #   {
+  #     'language' => 'go', 'completed_exercise' => 12,
+  #     'problems_count' => 30, 'language_name' => 'Go'
+  #   },
+  #   { 'language' => 'ruby', 'completed_exercise' => 60,
+  #     'problems_count' => 70, 'language_name' => 'Ruby'
+  #   }
+  # ]
+  def completed_exercises_by_language
+    ActiveRecord::Base.connection.execute(completed_count_sql).to_a.each do |hash|
```

Comment:
I believe that skipped exercises should count towards track completion, since if they skipped it they probably don't consider it a problem that they want to solve.